lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1559,8 +1559,8 @@ def plot_surface(self, X, Y, Z, *args, norm=None, vmin=None,
                 "Z contains NaN values. This may result in rendering "
                 "artifacts.")
 
-        # TODO: Support masked arrays
-        X, Y, Z = np.broadcast_arrays(X, Y, Z)
+        mask = np.ma.getmaskarray(Z)
```

Comment:
What happens if X or Y have a mask? My guess is we should take the union of all the masks to compute an 'overall' mask here.

Reply:
Good question, I see that [`pcolormesh`][1] doesn't accept masked `X` and `Y` values (but it looks like `pcolor` might). I've not tested `contour` but it [looks like masks are ignored on `X` and `Y` values][2]?

[1]: https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/axes/_axes.py#L5644
[2]: https://github.com/matplotlib/matplotlib/blob/81f0624c3baa7a2f4a01975fd9d71199894ab94d/lib/matplotlib/contour.py#L1462-L1464